mlflow/models/evaluation/default_evaluator.py

```diff
@@ -494,28 +499,51 @@ def _log_model_explainability(self):
             f: f2 for f, f2 in zip(self.feature_names, truncated_feature_names)
         }
 
-        sampled_X = shap.sample(self.X, sample_rows)
+        sampled_X = shap.sample(self.X, sample_rows, random_state=0)
+        if algorithm == "kernel":
+            background_X = shap.sample(self.X, sample_rows, random_state=3)
```

Comment:
`sampled_x` and `background_X` may have overlapping rows. Is that fine?

Reply:
I think it is fine.